### DISCUSSION TITLE
Width and Height should apply to all canvas and canvas styles

### DISCUSSION THREAD

**Discussion by ZeroRetro** [berviantoleo/react-multi-crop#1776] [Ideas]
I added `width={resizedWidth} height={resizedHeight}` to `ReactMultiCrop` and when I resize the component it seems that its only being changed in `lower-canvas` and not in the styles. 

The `canvas-wrapper` is changing because I also added `style={{ width: resizedWidth, height: resizedHeight }}` to the `ReactMultiCrop` component.

So i think that `width` and `height` parameters should also change the `upper-canvas` `width` and `height` and the styles for `canvas-container`, `lower-canvas` and `upper-canvas`

https://user-images.githubusercontent.com/33663801/234050337-2b3b371a-91e1-4376-821d-b8b2180da0c9.mov

**Comment by berviantoleo** [comment]
I'll look into that issue tomorrow. I'm not sure if I'll be able to fix that quickly.

**Reply by berviantoleo** [reply]
I made some changes [here](https://github.com/berviantoleo/react-multi-crop/pull/1778). I will need more time to ensure my changes don't break anything.

If @ZeroRetro has early feedback would be nice.

**Reply by ZeroRetro** [reply]
Thanks! Im starting to test. 

* Im not sure if its an issue caused by how Im managing the resize but as you can see when this happen the image is not changing its size. I think the image and crops should adjust to the new size, right? because both `canvas` are changing. Also you can see the image is being cropped when doing the size smaller and a weird white blink every time the size changes.

https://user-images.githubusercontent.com/33663801/234229417-ac07e03e-cc14-404b-af3a-7547e83b7798.mov

* If I just load another image with the new size then its getting its new size as expected.

https://user-images.githubusercontent.com/33663801/234231022-27fb9b2e-aa8c-48b2-a77e-4241ad6d6713.mov

I will let you know if I find anything useful

**Reply by berviantoleo** [reply]
Resizing the image will be hard because I will need to resize the crop box objects too. Resizing the crop box objects will be complex. I might need scaling instead of changing the actual width/height.

The main reason I provide the zoom, I don't want the image to fit with the canvas. I will need to change and check the scaling before generating the cropped images.

I will take a look, but I can't promise I can handle it.

**Reply by ZeroRetro** [reply]
I've got what I need by disabling the styles in all the elements below `ReactMultiCrop` and changing the `width` and `height` in all of them to `100%` still have an issue because the crop functionality got broken. I will keep trying. Also added `object-fit: contains` to the canvas that display the image

**Reply by ZeroRetro** [reply]
I have some news, Ive manage to get a good result only touching the `.css` and not the `width` and `height` properties of the component: 

https://user-images.githubusercontent.com/33663801/234837675-86cfa8d4-1d53-419d-a7ad-6f674da45c17.mov

This is what I need but now the `width` and `height` can only be redimensiones to a smaller size and not a bigger size, you can see some margins there when going bigger.

I fixed this by adding a fixed `width` and `height` properties to the component (in this example 3000 for both)

https://user-images.githubusercontent.com/33663801/234840163-2370c48b-43be-4b09-a72e-e83cd4a82c82.mov

As you can see the image is maintaining the aspect ratio and going bigger, but now the problems are with the crops, the borders and corners are changing according to the specified `width` and `height` I tried editing the `cornerSize` but I think I also need to edit the `borderSize`. Maybe you have better Idea of how to solve this.

Also I think would be really nice to add support to edit style for each crop, so we can have crops of different colours, something like:
`record={{ clippings: [{ id: 1, rect: { x1: 0.0, y1: 0.0, x2: 0.2, y2: 0.2 }, style: { ... }, rectPx: {} }] }}`

Maybe my `.css` and component files will be useful so Im sharing them here:
[ReactMultiCropTest.zip](https://github.com/berviantoleo/react-multi-crop/files/11342678/ReactMultiCropTest.zip)

**Reply by berviantoleo** [reply]
> Also I think would be really nice to add support to edit style for each crop, so we can have crops of different colours, something like:

I agree with this, but how will the user set the color? I mean, how will the user interact with the object and set the color?

> Maybe my .css and component files will be useful so Im sharing them here:

Hemm... Actually, I don't want to add any CSS for this component. I will check on that.

**Reply by ZeroRetro** [reply]
> I agree with this, but how will the user set the color? I mean, how will the user interact with the object and set the color?

That depends of how the developer wants to give the user access to that tool, for example I can add a `Select` with a few colours by my own under the `ReactMultiCrops` component and this will affect only to the selected crop and the new ones. 

https://user-images.githubusercontent.com/33663801/234908829-ee2e7c39-db37-49a3-8a21-3932d5a66349.mov

In this demo video im changing `borderColor` and `cropBackgroundColor` the second one you can see its not being affected in real time, only when the image changes. I think this will be fixed giving styles to each crop instead of all the component. 

> Hemm... Actually, I don't want to add any CSS for this component. I will check on that.

And about this, maybe you don't need to add any CSS to the component, but I would like to have more control about the crop styles.

Thank you for keeping an eye on your component! And for your fasts reply

**Reply by berviantoleo** [reply]
> In this demo video im changing borderColor and cropBackgroundColor the second one you can see its not being affected in real time, only when the image changes. I think this will be fixed giving styles to each crop instead of all the component.

Yeah, it's because I didn't change the color of each box after the color changes. It's related to my issue https://github.com/berviantoleo/react-multi-crop/issues/110. I notice that, but I still need to ensure my codes are not too messy. I need to handle that when the props change.

**Reply by berviantoleo** [reply]
You can check the newer version for changing crop colors. I made some changes to set the crop colors after the props changed.

**Reply by ZeroRetro** [reply]
Thanks! I will leave feedback when I have a chance

**Reply by ZeroRetro** [reply]
Finally got some time to test it again and yes, is working for all the crops, now will be great to edit style for each crop individually :eyes:

https://user-images.githubusercontent.com/33663801/235902753-4b3c9e30-1c8d-4fb9-82b0-a724a7484512.mov

**Reply by berviantoleo** [reply]
> now will be great to edit style for each crop individually

I'm thinking of that too. However, I will try on the weekend. I can't add the major changes on a weekday.

**Reply by ZeroRetro** [reply]
I think there is an issue when you have some crops defined in the component, this ones wont have color till the props refresh, the new ones will have the color tho. 

I mean the crops defined inside:
`record={{ clippings: [{ id: 1, rect: { x1: 0.0, y1: 0.0, x2: 0.2, y2: 0.2 }, rectPx: {} }] }}`

https://user-images.githubusercontent.com/33663801/236136701-73f9a5cc-dc2e-4da5-9c11-80631aa5b320.mov

**Reply by berviantoleo** [reply]
> I think there is an issue when you have some crops defined in the component, this ones wont have color till the props refresh, the new ones will have the color tho.

I added the quick fix for this. I didn't test it yet.

**Reply by ZeroRetro** [reply]
I've just tested it and seems to be working as expected, thanks for the quick response

**Reply by berviantoleo** [reply]
> now will be great to edit style for each crop individually

FYI, I made some changes regarding this.
